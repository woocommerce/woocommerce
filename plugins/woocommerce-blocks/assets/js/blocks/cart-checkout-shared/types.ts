export type ExpressCheckoutAttributes = {
	className?: string;
	buttonHeight: string;
	showButtonStyles: boolean;
	buttonBorderRadius: string;
	lock: {
		move: boolean;
		remove: boolean;
	};
};

export type ExpressCartAttributes = {
	className: string;
	buttonHeight: string;
	showButtonStyles: boolean;
	buttonBorderRadius: string;
};

export type ExpressPaymentSettings = {
	showButtonStyles: boolean;
	buttonHeight: string;
	buttonBorderRadius: string;
};
