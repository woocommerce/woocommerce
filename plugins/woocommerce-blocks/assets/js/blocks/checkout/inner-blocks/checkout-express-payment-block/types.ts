export type ExpressCheckoutAttributes = {
	className?: string;
	buttonHeight: string;
	showButtonStyles: boolean;
	buttonBorderRadius: string;
	lock: {
		move: boolean;
		remove: boolean;
	};
};
