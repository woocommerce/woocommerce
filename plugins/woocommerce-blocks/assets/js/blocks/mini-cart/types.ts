/**
 * External dependencies
 */
import { CartResponseTotals } from '@woocommerce/types';

export type IconType = 'cart' | 'bag' | 'bag-alt' | undefined;
export type productCountVisibilityType =
	| 'always'
	| 'never'
	| 'greater_than_zero'
	| undefined;

export interface ColorItem {
	color: string;
	name?: string;
	slug?: string;
	class?: string;
}
export interface BlockAttributes {
	initialCartItemsCount?: number;
	initialCartTotals?: CartResponseTotals;
	isInitiallyOpen?: boolean;
	colorClassNames?: string;
	style?: Record< string, Record< string, string > >;
	contents: string;
	miniCartIcon?: IconType;
	addToCartBehaviour: string;
	onCartClickBehaviour: string;
	hasHiddenPrice: boolean;
	priceColor: ColorItem;
	iconColor: ColorItem;
	productCountColor: ColorItem;
	productCountVisibility?: productCountVisibilityType;
}
