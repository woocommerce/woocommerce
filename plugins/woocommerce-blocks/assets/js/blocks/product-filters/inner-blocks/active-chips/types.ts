/**
 * External dependencies
 */
import { BlockEditProps } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { FilterBlockContext } from '../../types';

export type Color = {
	slug?: string;
	name?: string;
	class?: string;
	color: string;
};

export type BlockAttributes = {
	className: string;
	chipText?: string;
	customChipText?: string;
	chipBackground?: string;
	customChipBackground?: string;
	chipBorder?: string;
	customChipBorder?: string;
};

export type EditProps = BlockEditProps< BlockAttributes > & {
	style: Record< string, string >;
	context: FilterBlockContext;
	chipText: Color;
	setChipText: ( value: string ) => void;
	chipBackground: Color;
	setChipBackground: ( value: string ) => void;
	chipBorder: Color;
	setChipBorder: ( value: string ) => void;
};
