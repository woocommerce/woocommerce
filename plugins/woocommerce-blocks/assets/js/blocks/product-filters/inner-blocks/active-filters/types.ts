/**
 * External dependencies
 */
import { BlockEditProps } from '@wordpress/blocks';

export type BlockAttributes = {
	displayStyle: 'list' | 'chips';
};

export type EditProps = BlockEditProps< BlockAttributes >;
