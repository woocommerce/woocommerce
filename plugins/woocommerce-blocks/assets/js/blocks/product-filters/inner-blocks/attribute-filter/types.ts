/**
 * External dependencies
 */
import { BlockEditProps } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { sortOrders } from './constants';

export type BlockAttributes = {
	attributeId: number;
	showCounts: boolean;
	queryType: 'or' | 'and';
	displayStyle: string;
	selectType: string;
	isPreview: boolean;
	sortOrder: keyof typeof sortOrders;
	hideEmpty: boolean;
	clearButton: boolean;
};

export interface EditProps extends BlockEditProps< BlockAttributes > {
	debouncedSpeak: ( label: string ) => void;
}

type AttributeCount = {
	term: number;
	count: number;
};

export function isAttributeCounts(
	target: unknown
): target is AttributeCount[] {
	return (
		Array.isArray( target ) &&
		target.every( ( item ) => 'term' in item && 'count' in item )
	);
}
