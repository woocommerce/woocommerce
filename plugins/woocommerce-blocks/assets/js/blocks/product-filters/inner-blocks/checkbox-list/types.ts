/**
 * External dependencies
 */
import { BlockEditProps } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { FilterBlockContext } from '../../types';

export type Color = {
	slug: string;
	class: string;
	name: string;
	color: string;
};

export type BlockAttributes = {
	className: string;
	optionElementBorder: string;
	customOptionElementBorder: string;
	optionElementSelected: string;
	customOptionElementSelected: string;
	optionElement: string;
	customOptionElement: string;
};

export type EditProps = BlockEditProps< BlockAttributes > & {
	context: FilterBlockContext;
	optionElementBorder: Color;
	setOptionElementBorder: ( value: string ) => void;
	optionElementSelected: Color;
	setOptionElementSelected: ( value: string ) => void;
	optionElement: Color;
	setOptionElement: ( value: string ) => void;
};
