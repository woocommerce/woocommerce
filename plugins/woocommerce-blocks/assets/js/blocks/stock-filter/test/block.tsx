/**
 * External dependencies
 */
import React from '@wordpress/element';
import { act, render, screen, within, waitFor } from '@testing-library/react';
import { default as fetchMock } from 'jest-fetch-mock';
import userEvent from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import Block from '../block';
import { allSettings } from '../../../settings/shared/settings-init';
import { Attributes } from '../types';

const setWindowUrl = ( { url }: { url: string } ) => {
	Object.defineProperty( window, 'location', {
		value: {
			href: url,
		},
		writable: true,
	} );
};

const mockResults = {
	stock_status_counts: [
		{ status: 'instock', count: '18' },
		{ status: 'outofstock', count: '1' },
		{ status: 'onbackorder', count: '5' },
	],
};

jest.mock( '@woocommerce/base-context/hooks', () => {
	return {
		...jest.requireActual( '@woocommerce/base-context/hooks' ),
		useCollectionData: () => ( { isLoading: false, results: mockResults } ),
	};
} );

jest.mock( '@woocommerce/settings', () => {
	return {
		...jest.requireActual( '@woocommerce/settings' ),
		getSettingWithCoercion: jest
			.fn()
			.mockImplementation( ( key, defaultValue ) => {
				if ( key === 'hasFilterableProducts' ) {
					return true;
				}
				return defaultValue;
			} ),
	};
} );

type DisplayStyle = 'list' | 'dropdown';
type SelectType = 'single' | 'multiple';
interface SetupParams {
	filterStock?: string;
	displayStyle?: DisplayStyle;
	selectType?: SelectType;
	showCounts?: boolean;
	showFilterButton?: boolean;
}

const selectors = {
	list: '.wc-block-stock-filter.style-list',
	suggestionsContainer: '.components-form-token-field__suggestions-list',
	chipsContainer: '.components-form-token-field__token',
};

const setup = ( params: SetupParams = {} ) => {
	const url = `http://woo.local/${
		params.filterStock ? '?filter_stock_status=' + params.filterStock : ''
	}`;
	setWindowUrl( { url } );

	const attributes: Attributes = {
		displayStyle: params.displayStyle || 'list',
		selectType: params.selectType || 'single',
		showCounts: params.showCounts !== undefined ? params.showCounts : true,
		showFilterButton:
			params.showFilterButton !== undefined
				? params.showFilterButton
				: true,
		isPreview: false,
		heading: '',
		headingLevel: 3,
	};

	const { container, ...utils } = render(
		<Block attributes={ attributes } />,
		{ legacyRoot: true }
	);

	// We need to switch to React 17 rendering to allow these tests to keep passing, but as a result the React
	// rendering error will be shown.
	expect( console ).toHaveErroredWith(
		`Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot`
	);

	const getList = () => container.querySelector( selectors.list );
	const getDropdown = () => screen.queryByRole( 'combobox' );

	const getChipsContainers = () =>
		container.querySelectorAll( selectors.chipsContainer );
	const getSuggestionsContainer = () =>
		container.querySelector( selectors.suggestionsContainer );

	const getChips = ( value: string ) => {
		const chipsContainers = getChipsContainers();
		const chips = Array.from( chipsContainers ).find( ( chipsContainer ) =>
			chipsContainer
				? within( chipsContainer ).queryByText( value, {
						exact: false,
						ignore: '.components-visually-hidden',
				  } )
				: false
		);

		return chips || null;
	};
	const getSuggestion = ( value: string ) => {
		const suggestionsContainer = getSuggestionsContainer();
		if ( suggestionsContainer ) {
			return within( suggestionsContainer ).queryByText( value, {
				exact: false,
			} );
		}
		return null;
	};
	const getCheckbox = ( value: string ) => {
		const checkboxesContainer = getList();
		const checkboxes = checkboxesContainer
			? checkboxesContainer.querySelectorAll( 'input' )
			: [];

		const checkbox = Array.from( checkboxes ).find(
			( input ) => input.value === value
		);

		return checkbox;
	};

	const getRemoveButtonFromChips = ( chips: HTMLElement | null ) =>
		chips ? within( chips ).getByLabelText( 'Remove stock filter.' ) : null;

	const inStockLabel = 'In stock';
	const outOfStockLabel = 'Out of stock';
	const onBackstockLabel = 'On backorder';
	const inStockId = 'instock';
	const outOfStockId = 'outofstock';
	const onBackstoreId = 'onbackorder';

	const getInStockChips = () => getChips( inStockLabel );
	const getOutOfStockChips = () => getChips( outOfStockLabel );
	const getOnBackorderChips = () => getChips( onBackstockLabel );

	const getInStockSuggestion = () => getSuggestion( inStockLabel );
	const getOutOfStockSuggestion = () => getSuggestion( outOfStockLabel );
	const getOnBackorderSuggestion = () => getSuggestion( onBackstockLabel );

	const getInStockCheckbox = () => getCheckbox( inStockId );
	const getOutOfStockCheckbox = () => getCheckbox( outOfStockId );
	const getOnBackorderCheckbox = () => getCheckbox( onBackstoreId );

	return {
		...utils,
		container,
		getDropdown,
		getList,
		getInStockChips,
		getOutOfStockChips,
		getOnBackorderChips,
		getInStockSuggestion,
		getOutOfStockSuggestion,
		getOnBackorderSuggestion,
		getInStockCheckbox,
		getOutOfStockCheckbox,
		getOnBackorderCheckbox,
		getRemoveButtonFromChips,
	};
};

interface SetupParams {
	filterStock?: string;
	displayStyle?: DisplayStyle;
	selectType?: SelectType;
}

const setupSingleChoiceList = ( filterStock = 'instock' ) =>
	setup( {
		filterStock,
		displayStyle: 'list',
		selectType: 'single',
	} );

const setupMultipleChoiceList = ( filterStock = 'instock' ) =>
	setup( {
		filterStock,
		displayStyle: 'list',
		selectType: 'multiple',
	} );

const setupSingleChoiceDropdown = ( filterStock = 'instock' ) =>
	setup( {
		filterStock,
		displayStyle: 'dropdown',
		selectType: 'single',
	} );

const setupMultipleChoiceDropdown = ( filterStock = 'instock' ) =>
	setup( {
		filterStock,
		displayStyle: 'dropdown',
		selectType: 'multiple',
	} );

describe( 'Filter by Stock block', () => {
	beforeEach( () => {
		allSettings.stockStatusOptions = {
			instock: 'In stock',
			outofstock: 'Out of stock',
			onbackorder: 'On backorder',
		};
	} );

	afterEach( () => {
		fetchMock.resetMocks();
	} );

	it( 'renders the stock filter block', async () => {
		const { container } = setup( {
			showFilterButton: false,
			showCounts: false,
		} );
		expect( container ).toMatchSnapshot();
	} );

	it( 'renders the stock filter block with the filter button', async () => {
		const { container } = setup( {
			showFilterButton: true,
			showCounts: false,
		} );
		expect( container ).toMatchSnapshot();
	} );

	it( 'renders the stock filter block with the product counts', async () => {
		const { container } = setup( {
			showFilterButton: false,
			showCounts: true,
		} );
		expect( container ).toMatchSnapshot();
	} );

	describe( 'Single choice Dropdown', () => {
		test( 'renders dropdown', () => {
			const { getDropdown, getList } = setupSingleChoiceDropdown();
			expect( getDropdown() ).toBeInTheDocument();
			expect( getList() ).toBeNull();
		} );

		test( 'renders chips based on URL params', () => {
			const ratingParam = 'instock';
			const { getInStockChips, getOutOfStockChips, getOnBackorderChips } =
				setupSingleChoiceDropdown( ratingParam );

			expect( getInStockChips() ).toBeInTheDocument();
			expect( getOutOfStockChips() ).toBeNull();
			expect( getOnBackorderChips() ).toBeNull();
		} );

		test( 'replaces chosen option when another one is clicked', async () => {
			const user = userEvent.setup();
			const ratingParam = 'instock';
			const {
				getDropdown,
				getInStockChips,
				getOutOfStockChips,
				getOutOfStockSuggestion,
			} = setupSingleChoiceDropdown( ratingParam );

			expect( getInStockChips() ).toBeInTheDocument();
			expect( getOutOfStockChips() ).toBeNull();

			const dropdown = getDropdown();

			if ( dropdown ) {
				await act( async () => {
					await user.click( dropdown );
				} );
			}

			const outOfStockSuggestion = getOutOfStockSuggestion();

			if ( outOfStockSuggestion ) {
				await act( async () => {
					await user.click( outOfStockSuggestion );
				} );
			}

			expect( getInStockChips() ).toBeNull();
			expect( getOutOfStockChips() ).toBeInTheDocument();
		} );

		test( 'removes the option when the X button is clicked', async () => {
			const user = userEvent.setup();
			const ratingParam = 'outofstock';
			const {
				getInStockChips,
				getOutOfStockChips,
				getOnBackorderChips,
				getRemoveButtonFromChips,
			} = setupMultipleChoiceDropdown( ratingParam );

			await waitFor( () => {
				expect( getInStockChips() ).toBeNull();
				expect( getOutOfStockChips() ).toBeInTheDocument();
				expect( getOnBackorderChips() ).toBeNull();
			} );

			const removeOutOfStockButton = getRemoveButtonFromChips(
				getOutOfStockChips()
			);

			if ( removeOutOfStockButton ) {
				act( async () => {
					await user.click( removeOutOfStockButton );
				} );
			}

			await waitFor( () => {
				expect( getInStockChips() ).toBeNull();
				expect( getOutOfStockChips() ).toBeNull();
				expect( getOnBackorderChips() ).toBeNull();
			} );
		} );
	} );

	describe( 'Multiple choice Dropdown', () => {
		test( 'renders dropdown', () => {
			const { getDropdown, getList } = setupMultipleChoiceDropdown();
			expect( getDropdown() ).toBeDefined();
			expect( getList() ).toBeNull();
		} );

		test( 'renders chips based on URL params', () => {
			const ratingParam = 'instock,onbackorder';
			const { getInStockChips, getOutOfStockChips, getOnBackorderChips } =
				setupMultipleChoiceDropdown( ratingParam );

			expect( getInStockChips() ).toBeInTheDocument();
			expect( getOutOfStockChips() ).toBeNull();
			expect( getOnBackorderChips() ).toBeInTheDocument();
		} );

		test( 'adds chosen option to another one that is clicked', async () => {
			const user = userEvent.setup();
			const ratingParam = 'onbackorder';
			const {
				getDropdown,
				getInStockChips,
				getOutOfStockChips,
				getOnBackorderChips,
				getInStockSuggestion,
				getOutOfStockSuggestion,
			} = setupMultipleChoiceDropdown( ratingParam );

			await waitFor( () => {
				expect( getInStockChips() ).toBeNull();
				expect( getOutOfStockChips() ).toBeNull();
				expect( getOnBackorderChips() ).toBeInTheDocument();
			} );
			const dropdown = getDropdown();

			if ( dropdown ) {
				await act( async () => {
					await user.click( dropdown );
				} );
			}

			const inStockSuggestion = getInStockSuggestion();

			if ( inStockSuggestion ) {
				await act( async () => {
					await user.click( inStockSuggestion );
				} );
			}

			expect( getInStockChips() ).toBeInTheDocument();
			expect( getOutOfStockChips() ).toBeNull();
			expect( getOnBackorderChips() ).toBeInTheDocument();

			const freshDropdown = getDropdown();
			if ( freshDropdown ) {
				await act( async () => {
					await user.click( freshDropdown );
				} );
			}

			const outOfStockSuggestion = getOutOfStockSuggestion();

			if ( outOfStockSuggestion ) {
				userEvent.click( outOfStockSuggestion );
			}

			await waitFor( () => {
				expect( getInStockChips() ).toBeInTheDocument();
				expect( getOutOfStockChips() ).toBeInTheDocument();
				expect( getOnBackorderChips() ).toBeInTheDocument();
			} );
		} );

		test( 'removes the option when the X button is clicked', async () => {
			const user = userEvent.setup();
			const ratingParam = 'instock,outofstock,onbackorder';
			const {
				getInStockChips,
				getOutOfStockChips,
				getOnBackorderChips,
				getRemoveButtonFromChips,
			} = setupMultipleChoiceDropdown( ratingParam );

			await waitFor( () => {
				expect( getInStockChips() ).toBeInTheDocument();
				expect( getOutOfStockChips() ).toBeInTheDocument();
				expect( getOnBackorderChips() ).toBeInTheDocument();
			} );

			const removeOutOfStockButton = getRemoveButtonFromChips(
				getOutOfStockChips()
			);

			if ( removeOutOfStockButton ) {
				act( async () => {
					await user.click( removeOutOfStockButton );
				} );
			}

			await waitFor( () => {
				expect( getInStockChips() ).toBeInTheDocument();
				expect( getOutOfStockChips() ).toBeNull();
				expect( getOnBackorderChips() ).toBeInTheDocument();
			} );
		} );
	} );

	describe( 'Single choice List', () => {
		test( 'renders list', () => {
			const { getDropdown, getList } = setupSingleChoiceList();
			expect( getDropdown() ).toBeNull();
			expect( getList() ).toBeInTheDocument();
		} );

		test( 'renders checked options based on URL params', () => {
			const ratingParam = 'instock';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupSingleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeTruthy();
			expect( getOutOfStockCheckbox()?.checked ).toBeFalsy();
			expect( getOnBackorderCheckbox()?.checked ).toBeFalsy();
		} );

		test( 'replaces chosen option when another one is clicked', async () => {
			const user = userEvent.setup();
			const ratingParam = 'outofstock';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupSingleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeFalsy();
			expect( getOutOfStockCheckbox()?.checked ).toBeTruthy();
			expect( getOnBackorderCheckbox()?.checked ).toBeFalsy();

			const onBackorderCheckbox = getOnBackorderCheckbox();

			if ( onBackorderCheckbox ) {
				await act( async () => {
					await user.click( onBackorderCheckbox );
				} );
			}

			expect( getInStockCheckbox()?.checked ).toBeFalsy();
			expect( getOutOfStockCheckbox()?.checked ).toBeFalsy();
			expect( getOnBackorderCheckbox()?.checked ).toBeTruthy();
		} );

		test( 'removes the option when it is clicked again', async () => {
			const ratingParam = 'onbackorder';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupMultipleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeFalsy();
			expect( getOutOfStockCheckbox()?.checked ).toBeFalsy();
			expect( getOnBackorderCheckbox()?.checked ).toBeTruthy();

			const onBackorderCheckbox = getOnBackorderCheckbox();

			if ( onBackorderCheckbox ) {
				userEvent.click( onBackorderCheckbox );
			}

			await waitFor( () => {
				expect( getInStockCheckbox()?.checked ).toBeFalsy();
				expect( getOutOfStockCheckbox()?.checked ).toBeFalsy();
				expect( getOnBackorderCheckbox()?.checked ).toBeFalsy();
			} );
		} );
	} );

	describe( 'Multiple choice List', () => {
		test( 'renders list', () => {
			const { getDropdown, getList } = setupMultipleChoiceList();
			expect( getDropdown() ).toBeNull();
			expect( getList() ).toBeInTheDocument();
		} );

		test( 'renders chips based on URL params', () => {
			const ratingParam = 'instock,onbackorder';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupMultipleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeTruthy();
			expect( getOutOfStockCheckbox()?.checked ).toBeFalsy();
			expect( getOnBackorderCheckbox()?.checked ).toBeTruthy();
		} );

		test( 'adds chosen option to another one that is clicked', async () => {
			const ratingParam = 'outofstock,onbackorder';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupMultipleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeFalsy();
			expect( getOutOfStockCheckbox()?.checked ).toBeTruthy();
			expect( getOnBackorderCheckbox()?.checked ).toBeTruthy();

			const inStockCheckbox = getInStockCheckbox();

			if ( inStockCheckbox ) {
				userEvent.click( inStockCheckbox );
			}

			await waitFor( () => {
				expect( getInStockCheckbox()?.checked ).toBeTruthy();
				expect( getOutOfStockCheckbox()?.checked ).toBeTruthy();
				expect( getOnBackorderCheckbox()?.checked ).toBeTruthy();
			} );
		} );

		test( 'removes the option when it is clicked again', async () => {
			const ratingParam = 'instock,outofstock';
			const {
				getInStockCheckbox,
				getOutOfStockCheckbox,
				getOnBackorderCheckbox,
			} = setupMultipleChoiceList( ratingParam );

			expect( getInStockCheckbox()?.checked ).toBeTruthy();
			expect( getOutOfStockCheckbox()?.checked ).toBeTruthy();
			expect( getOnBackorderCheckbox()?.checked ).toBeFalsy();

			const inStockCheckbox = getInStockCheckbox();

			if ( inStockCheckbox ) {
				userEvent.click( inStockCheckbox );
			}

			await waitFor( () => {
				expect( getInStockCheckbox()?.checked ).toBeFalsy();
				expect( getOutOfStockCheckbox()?.checked ).toBeTruthy();
				expect( getOnBackorderCheckbox()?.checked ).toBeFalsy();
			} );
		} );
	} );
} );
