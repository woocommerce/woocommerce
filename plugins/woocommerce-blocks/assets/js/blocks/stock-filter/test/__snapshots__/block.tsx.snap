// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter by Stock block renders the stock filter block 1`] = `
<div>
  <div
    class="wc-block-stock-filter style-list"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-0"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-0"
              type="checkbox"
              value="instock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-1"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-1"
              type="checkbox"
              value="onbackorder"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-2"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-2"
              type="checkbox"
              value="outofstock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  />
</div>
`;

exports[`Filter by Stock block renders the stock filter block with the filter button 1`] = `
<div>
  <div
    class="wc-block-stock-filter style-list"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-3"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-3"
              type="checkbox"
              value="instock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-4"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-4"
              type="checkbox"
              value="onbackorder"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-5"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-5"
              type="checkbox"
              value="outofstock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  >
    <button
      class="wp-block-button__link wc-block-filter-submit-button wc-block-components-filter-submit-button wc-block-stock-filter__button"
      type="submit"
    >
      <span
        aria-hidden="true"
      >
        Apply
      </span>
      <span
        class="screen-reader-text"
      >
        Apply filter
      </span>
    </button>
  </div>
</div>
`;

exports[`Filter by Stock block renders the stock filter block with the product counts 1`] = `
<div>
  <div
    class="wc-block-stock-filter style-list"
  >
    <ul
      class="wc-block-checkbox-list wc-block-components-checkbox-list wc-block-stock-filter-list"
    >
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-6"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-6"
              type="checkbox"
              value="instock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              In stock
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  18
                </span>
                <span
                  class="screen-reader-text"
                >
                  18 products
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-7"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-7"
              type="checkbox"
              value="onbackorder"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              On backorder
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  5
                </span>
                <span
                  class="screen-reader-text"
                >
                  5 products
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
      <li>
        <div
          class="wc-block-components-checkbox wc-block-checkbox-list__checkbox"
        >
          <label
            for="wc-block-checkbox-list-option-8"
          >
            <input
              aria-invalid="false"
              class="wc-block-components-checkbox__input"
              id="wc-block-checkbox-list-option-8"
              type="checkbox"
              value="outofstock"
            />
            <svg
              aria-hidden="true"
              class="wc-block-components-checkbox__mark"
              viewBox="0 0 24 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
              />
            </svg>
            <span
              class="wc-block-components-checkbox__label"
            >
              Out of stock
              <span
                class="wc-filter-element-label-list-count"
              >
                <span
                  aria-hidden="true"
                >
                  1
                </span>
                <span
                  class="screen-reader-text"
                >
                  1 product
                </span>
              </span>
            </span>
          </label>
        </div>
      </li>
    </ul>
  </div>
  <div
    class="wc-block-stock-filter__actions"
  />
</div>
`;
